frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -343,60 +363,51 @@ func (m *CVDToolInstanceManager) launchFromAndroidCI(
 	if merr != nil {
 		return nil, merr
 	}
-	instanceNumber := atomic.AddUint32(&m.instanceCounter, 1)
-	cvdName := fmt.Sprintf("cvd-%d", instanceNumber)
-	runtimeDir := m.paths.RuntimesRootDir + "/" + cvdName
-	if err := createNewDir(runtimeDir); err != nil {
-		return nil, err
-	}
 	startParams := startCVDParams{
-		InstanceNumber:   instanceNumber,
+		InstanceNumbers:  m.newInstanceNumbers(instancesCount),
 		MainArtifactsDir: mainBuildDir,
-		RuntimeDir:       runtimeDir,
+		RuntimeDir:       m.paths.RuntimesRootDir,
 		KernelDir:        kernelBuildDir,
 		BootloaderDir:    bootloaderBuildDir,
 	}
 	if err := m.startCVDHandler.Start(startParams); err != nil {
 		return nil, err
 	}
-	cvd := &apiv1.CVD{}
-	*cvd = *req.CVD
-	cvd.Name = cvdName
 	// TODO: Remove once `acloud CLI` gets deprecated.
-	if cvdName == "cvd-1" {
-		go runAcloudSetup(m.execContext, m.paths.ArtifactsRootDir, mainBuildDir, runtimeDir)
+	if contains(startParams.InstanceNumbers, 1) {
```

Comment:
nit: This is common to all `launch...` functions, it could be done in the calling function instead.

Reply:
good observation, however the acloud script requires the artifact directory which is computed differently in each of the `launch` functions which makes it not trivial to move it to the calling function.